# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1456955183 18000
#      Wed Mar 02 16:46:23 2016 -0500
# Node ID df859661e31e596953b10e8748a8270b6fee8335
# Parent  309bba07a4f6acb8edc094942162889449434f31
generate default-qt-settings at configure time

* acinclude.m4 (OCTAVE_CONFIG_MOVE_IF_CHANGE_FILES): New macro.
* configure.ac: Use it.
* mk-default-qt-settings.in: New script template created from
do_subst_qt_settings macro in common.mk
* common.mk (do_subst_qt_settings): Delete macro definition.

diff --git a/build-aux/common.mk b/build-aux/common.mk
--- a/build-aux/common.mk
+++ b/build-aux/common.mk
@@ -503,23 +503,16 @@ define do_subst_script_vals
     -e "s|%FIND%|${FIND}|g" \
     -e "s|%SED%|${SED}|g" \
     -e "s|%ADDRESS_SANITIZER_OPTIONS%|${ADDRESS_SANITIZER_OPTIONS}|g" \
     -e "s|%abs_top_srcdir%|${abs_top_srcdir}|" \
     -e "s|%builddir%|$(shell pwd)|" > $@-t && \
   $(simple_move_if_change_rule)
 endef
 
-define do_subst_qt_settings
-  $(SED) < $< \
-    -e "s|%DEFAULT_TERMINAL_FONT%|${DEFAULT_TERMINAL_FONT}|" \
-    -e "s|%DEFAULT_TERMINAL_FONT_SIZE%|${DEFAULT_TERMINAL_FONT_SIZE}|" > $@-t && \
-  $(simple_move_if_change_rule)
-endef
-
 define subst-bison-api-decls
   case "$(BISON_API_PREFIX_DECL_STYLE)" in \
     *api*) \
       case "$(BISON_API_PREFIX_DECL_STYLE)" in \
        *brace*) \
          api_prefix_decl='%define api.prefix {$(1)}'; ;; \
        *) \
          api_prefix_decl='%define api.prefix "$(1)"'; ;; \
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -3191,17 +3191,20 @@ AC_SUBST(ac_config_headers)
 
 dnl We use a .in.h file for oct-conf-post.h simply to copy it to
 dnl the build tree and so that we don't have to add the -I$(top_srcdir)
 dnl to any CPPFLAGS variables.
 AC_CONFIG_FILES([
   oct-conf-post.h:oct-conf-post.in.h
   Makefile
   doc/doxyhtml/Doxyfile
-  libgnu/Makefile
+  libgnu/Makefile])
+
+OCTAVE_CONFIG_MOVE_IF_CHANGE_FILES([
+  libgui/mk-default-qt-settings
   liboctave/cruft/mkf77def])
 
 AC_OUTPUT
 
 ### Print a summary so that important information isn't missed.
 
 AC_MSG_NOTICE([
 
diff --git a/libgui/mk-default-qt-settings.in b/libgui/mk-default-qt-settings.in
new file mode 100644
--- /dev/null
+++ b/libgui/mk-default-qt-settings.in
@@ -0,0 +1,28 @@
+#! /bin/sh
+#
+# Copyright (C) 2016 John W. Eaton
+#
+# This file is part of Octave.
+#
+# Octave is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at
+# your option) any later version.
+#
+# Octave is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Octave; see the file COPYING.  If not, see
+# <http://www.gnu.org/licenses/>.
+
+SED=${SED:-sed}
+
+DEFAULT_TERMINAL_FONT = @DEFAULT_TERMINAL_FONT@
+DEFAULT_TERMINAL_FONT_SIZE = @DEFAULT_TERMINAL_FONT_SIZE@
+
+$(SED) \
+  -e "s|%DEFAULT_TERMINAL_FONT%|${DEFAULT_TERMINAL_FONT}|" \
+  -e "s|%DEFAULT_TERMINAL_FONT_SIZE%|${DEFAULT_TERMINAL_FONT_SIZE}|"
diff --git a/libgui/module.mk b/libgui/module.mk
--- a/libgui/module.mk
+++ b/libgui/module.mk
@@ -65,18 +65,24 @@ libgui_liboctgui_la_LDFLAGS = \
   -bindir $(bindir) \
   $(LIBOCTGUI_LINK_OPTS) \
   $(WARN_LDFLAGS)
 
 octetc_DATA += libgui/default-qt-settings
 
 octlocale_DATA += $(LOCALES)
 
-libgui/default-qt-settings: libgui/default-qt-settings.in | libgui/$(octave_dirstamp)
-	$(AM_V_GEN)$(do_subst_qt_settings)
+define gen-default-qt-settings
+  rm -f $@-t $@ && \
+  $(SHELL) libgui/mk-default-qt-settings < $< > $@-t && \
+  mv $@-t $@
+endef
+
+libgui/default-qt-settings: libgui/default-qt-settings.in libgui/mk-default-qt-settings | libgui/$(octave_dirstamp)
+	$(AM_V_GEN)$(gen-default_qt_settings)
 
 DIRSTAMP_FILES += \
   libgui/$(octave_dirstamp)
 
 define moc-command
   rm -f $@-t $@ && \
   ( echo "#ifdef HAVE_CONFIG_H"; \
     echo '#include "config.h"'; \
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -1312,16 +1312,26 @@ dnl
 AC_DEFUN([OCTAVE_CONFIGURE_WARNING_SUMMARY], [
   m4_set_foreach([summary_warning_list], [elt], [
     if test -n "[$]elt"; then
       AC_MSG_WARN([$]elt)
       warn_msg_printed=true
     fi])
 ])
 dnl
+dnl Like AC_CONFIG_FILES, but don't touch the output file if it already
+dnl exists and hasn't changed.
+dnl
+AC_DEFUN([OCTAVE_CONFIG_MOVE_IF_CHANGE_FILES], [
+  m4_foreach_w([elt], [$1], [
+    AC_CONFIG_FILES(elt[-tmp:]elt[.in])
+    AC_CONFIG_COMMANDS(elt,
+    [$SHELL $srcdir/build-aux/move-if-change ]elt[-tmp ]elt)])])
+
+dnl
 dnl Check if the C++ library has the bit_and, bit_or, and bit_xor
 dnl templates defined.
 dnl
 AC_DEFUN([OCTAVE_CXX_BITWISE_OP_TEMPLATES], [
   AC_CACHE_CHECK([whether bit_and, bit_or, bit_xor are defined in the C++ library],
     [octave_cv_cxx_bitwise_op_templates],
     [AC_LANG_PUSH(C++)
     AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
